test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -4021,5 +4021,236 @@ public void PackCommand_Deterministic_MultiplePackInvocations_CreateIdenticalPac
             }
         }
 
+        [PlatformFact(Platform.Windows)]
+        public void PackCommand_PackageIcon_HappyPath_Suceeds()
+        {
+            var testDirBuilder = TestDirectoryBuilder.Create();
+            var projectBuilder = ProjectFileBuilder.Create();
+
+            testDirBuilder
+                .WithFile("test\\icon.jpg", 10);
+
+            projectBuilder
+                .WithProjectName("test")
+                .WithPackageIcon("content\\icon.jpg")
+                .WithItem("None", "icon.jpg", null);
+
+            using (var srcDir = testDirBuilder.Build())
+            {
+                projectBuilder.Build(msbuildFixture, srcDir.Path);
+                var result = msbuildFixture.PackProject(projectBuilder.ProjectFolder, projectBuilder.ProjectName, string.Empty);
+
+                ValidatePackIcon(projectBuilder);
+            }
+        }
+
+        [PlatformFact(Platform.Windows)]
+        public void PackCommand_PackageIcon_ImplicitFile_Suceeds()
```

Comment:
What's the difference between PackCommand_PackageIcon_ImplicitFile_Suceeds and PackCommand_PackageIcon_FolderNested_Suceeds? 

Feels like they are testing the same thing. 

Note the reason why I'm focusing on these tests is because they each take about 10-15s to run and that quickly adds up when you account for the full test suite.

Reply:
Folder nested: testing `folder\**` don't change the folder layout inside the package
implicit file: testing `folder\**` sightly the same but, testing the root folder.

Reply:
I'd say that's not relevant for icons. 

If there's issues with the nesting and layout preservation, then it's a content include pack problem, not an icon one. 

You should test the icon specific scenarios. 
If there's a missing test for this scenario, add it in that context.

Reply:
Ok, I will mix both tests into one test.